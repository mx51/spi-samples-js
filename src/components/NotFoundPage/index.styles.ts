import { createStyles, makeStyles, Theme } from '@material-ui/core';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      minHeight: `calc(100vh - 48px)`,
    },
    title: {
      fontSize: '2rem',
      marginTop: theme.spacing(4),
      marginBottom: theme.spacing(1),
    },
    subtitle: {
      marginBottom: theme.spacing(4),
    },
  })
);
export default useStyles;
