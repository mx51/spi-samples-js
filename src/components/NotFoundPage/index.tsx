import Typography from '@material-ui/core/Typography';
import React from 'react';
import { Link } from 'react-router-dom';
// component
import Layout from '../Layout';

const NotFound: React.FC = () => (
  <Layout>
    <Typography>404 Error Found ..</Typography>
    <Typography>
      Back to <Link to="/">Home</Link>
    </Typography>
  </Layout>
);

export default NotFound;
