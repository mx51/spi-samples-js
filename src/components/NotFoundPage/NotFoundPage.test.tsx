import React from 'react';
import { cleanup } from '@testing-library/react';
import NotFoundPage from '.';
import mockWithRedux from '../../utils/tests/common';

describe('Test <NotFoundPage />', () => {
  afterEach(cleanup);

  test('should match NotFoundPage snapshot test', () => {
    // Arrange
    const container = mockWithRedux(<NotFoundPage />);

    // Assert
    expect(container).toMatchSnapshot();
  });
});
