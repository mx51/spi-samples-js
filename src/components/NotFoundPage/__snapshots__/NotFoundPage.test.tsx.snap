// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <NotFoundPage /> should match NotFoundPage snapshot test 1`] = `
<div>
  <div>
    <div
      class="makeStyles-root-4"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-navbar-5 mui-fixed MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
        >
          <div
            class="MuiBox-root MuiBox-root-10 makeStyles-navbar-7"
          >
            <div
              class="MuiBox-root MuiBox-root-11"
            >
              <button
                aria-label="menu"
                class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-8 MuiIconButton-colorInherit MuiIconButton-edgeStart"
                data-test-id="navbarMenuIcon"
                name="menuIcon"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <svg
                height="15"
                width="48"
              >
                PrimaryLogoIconWhite.svg
              </svg>
            </div>
          </div>
        </div>
      </header>
    </div>
    <div
      class="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
    >
      <svg>
        NotFoundImage.svg
      </svg>
      <h1
        class="MuiTypography-root makeStyles-title-2 MuiTypography-h6"
      >
        Page not found
      </h1>
      <p
        class="MuiTypography-root makeStyles-subtitle-3 MuiTypography-body2"
      >
        The page you are looking for doesnâ€™t exist or another error occurred.
      </p>
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        href="/"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiButton-label"
        >
          Go to Sample POS
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</div>
`;
